src/librustc_typeck/check/generator_interior.rs

```diff
@@ -14,7 +14,7 @@ use crate::util::nodemap::FxHashMap;
 
 struct InteriorVisitor<'a, 'tcx> {
     fcx: &'a FnCtxt<'a, 'tcx>,
-    types: FxHashMap<Ty<'tcx>, usize>,
+    types: FxHashMap<(Ty<'tcx>, Span, Option<Span>), usize>,
```

Comment:
Can we document what the heck is going on in this tuple?

Or, better yet, introduce a named struct so the fields can have meaningful names?

Reply:
If we did that, we could also use the type in the `TypeckTables`

Reply:
I've added a type and used it in `TypeckTables` too, not sure about the name.